### PR TITLE
[Enhancement] Optimize multiple like predicates

### PR DESCRIPTION
**Pull request by satanson**
## Why I'm doing:

1. Multiple like predicates with the same column can converted into a regex predicate.
e.g: 
```SQL
region like '%ABC' or region like 'DEF_G'
```
can converted to 
```SQL
region REGEXP '^((.*ABC)|(DEF.G))$'
```
The latter would be more efficient.

2. Case-when simplification in previous versions is only applied to scan operator, it can applied to predicates of other operators, too.

## What I'm doing:
1. Introduce a ScalarOparatorRewriteRule to optimize multiple like predicate;
2. Introduce a SimplifyCaseWhenPredicateRule to travel the plan tree to process prediates of each logical operator. 
Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.3
  - [x] 3.2
  - [x] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by stdpain** [comment]
Please add the session variable to this rule. (This optimization may not always be a positive optimization). Provide the performace case if you can.

**Comment by ZiheLiu** [comment]
Since we optimize the logic of some special simple like predicates including `%xxx`, `xxx%`, `%xxx%`, and `xxx`, is it possible that sometimes two simple like predicates are better than one complex regex predicate?

**Comment by satanson** [comment]
> Please add the session variable to this rule. (This optimization may not always be a positive optimization). Provide the performace case if you can.

@stdpain session variables is not required,  I will conduct some tests on it to find out a threshold, only the number of above this threshold  t

**Comment by satanson** [comment]
> Since we optimize the logic of some special simple like predicates including `%xxx`, `xxx%`, `%xxx%`, and `xxx`, is it possible that sometimes two simple like predicates are better than one complex regex predicate?

some tests would be conducted on this feature

**Review by kangkaisen — APPROVED** [review]

**Review by LiShuMing — APPROVED** [review]

**Review by ZiheLiu — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ConsolidateLikesRule.java [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class ConsolidateLikesRule extends TopDownScalarOperatorRewriteRule {
```
Comment (packy92) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rewrite/scalar/ConsolidateLikesRule.java:41]: please add some comment in the future.

**Review by packy92 — APPROVED** [review]

**Review by Seaven — APPROVED** [review]

**Pull request merged**